x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/legacy_create_legacy_notification.ts

```diff
@@ -0,0 +1,110 @@
+// eslint-disable-next-line no-restricted-imports
+import { legacyReadNotifications } from '../../notifications/legacy_read_notifications';
+// eslint-disable-next-line no-restricted-imports
+import { LegacyRuleNotificationAlertTypeParams } from '../../notifications/legacy_types';
+// eslint-disable-next-line no-restricted-imports
+import { legacyAddTags } from '../../notifications/legacy_add_tags';
+// eslint-disable-next-line no-restricted-imports
+import { legacyCreateNotifications } from '../../notifications/legacy_create_notifications';
+
+/**
+ * Given an "alert_id" and a valid "action_id" this will create a legacy notification. This is for testing
+ * purposes only and should not be used for production. It is behind a route with the words "internal" and
+ * "legacy" to announce its legacy and internal intent.
+ * @deprecated Once we no longer have legacy notifications and "side car actions" this can be removed.
+ * @param router The router
+ */
+export const legacyCreateLegacyNotificationRoute = (router: SecuritySolutionPluginRouter): void => {
+  router.post(
+    {
+      path: '/internal/api/detection/legacy/notifications',
+      validate: {
+        query: schema.object({ alert_id: schema.string() }),
+        body: schema.object({
+          name: schema.string(),
+          interval: schema.string(),
+          actions: schema.arrayOf(
+            schema.object({
+              id: schema.string(),
+              group: schema.string(),
+              params: schema.object({
+                message: schema.string(),
+              }),
+              actionTypeId: schema.string(),
+            })
+          ),
+        }),
+      },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const rulesClient = context.alerting.getRulesClient();
+      const savedObjectsClient = context.core.savedObjects.client;
+      const { alert_id: ruleAlertId } = request.query;
+      const { actions, interval, name } = request.body;
+      try {
+        // This is to ensure it exists before continuing.
+        await rulesClient.get({ id: ruleAlertId });
+        const notification = await legacyReadNotifications({
+          rulesClient,
+          id: undefined,
+          ruleAlertId,
+        });
+        if (notification != null) {
+          await rulesClient.update<LegacyRuleNotificationAlertTypeParams>({
+            id: notification.id,
+            data: {
+              tags: legacyAddTags([], ruleAlertId),
+              name,
+              schedule: {
+                interval,
+              },
+              actions,
+              params: {
+                ruleAlertId,
+              },
+              throttle: null,
+              notifyWhen: null,
+            },
+          });
+        } else {
+          await legacyCreateNotifications({
+            rulesClient,
+            actions: [],
```

Comment:
small fix for not defaulting to an empty actions array when the notification is null.

```suggestion
            actions,
```